src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,444 @@
+from redis import RedisError
+
+from nv_ingest.framework.schemas.framework_message_wrapper_schema import MessageWrapper
+from nv_ingest.framework.util.service.impl.ingest.redis_ingest_service import RedisIngestService
+from nv_ingest.framework.util.service.meta.ingest.ingest_service_meta import IngestServiceMeta
+from nv_ingest_api.util.service_clients.client_base import FetchMode
+from nv_ingest_client.primitives.jobs.job_spec import JobSpec
+from nv_ingest_client.primitives.tasks.extract import ExtractTask
+
+# For PDF splitting
+import pypdfium2 as pdfium
+
+logger = logging.getLogger("uvicorn")
+tracer = trace.get_tracer(__name__)
+
+router = APIRouter()
+
+# Reuse V1 state constants and helper functions
+from ..v1.ingest import (
+    _get_ingest_service,
+    trace_id_to_uuid,
+    INGEST_SERVICE_T,
+    STATE_RETRIEVED_DESTRUCTIVE,
+    STATE_RETRIEVED_NON_DESTRUCTIVE,
+    STATE_RETRIEVED_CACHED,
+    STATE_FAILED,
+    STATE_PROCESSING,
+    STATE_SUBMITTED,
+    INTERMEDIATE_STATES,
+)
+
+
+def split_pdf_to_pages(pdf_content: bytes) -> List[bytes]:
+    """
+    Split a PDF into individual pages using pypdfium2.
+    
+    Parameters
+    ----------
+    pdf_content : bytes
+        The PDF file content as bytes
+        
+    Returns
+    -------
+    List[bytes]
+        List of single-page PDFs as bytes
+    """
+    page_pdfs = []
+    
+    # Load the PDF
+    pdf = pdfium.PdfDocument(pdf_content)
+    page_count = len(pdf)
+    
+    for page_index in range(page_count):
+        # Create a new document with just this page
+        new_pdf = pdfium.PdfDocument.new()
+        new_pdf.import_pages(pdf, [page_index])
+        
+        # Save to BytesIO buffer
+        buffer = BytesIO()
+        new_pdf.save(buffer)
+        buffer.seek(0)
+        page_pdfs.append(buffer.read())
+        
+        # Clean up the new PDF
+        new_pdf.close()
+    
+    # Clean up original PDF
+    pdf.close()
+    
+    return page_pdfs
+
+
+def get_pdf_page_count(pdf_content: bytes) -> int:
+    """Get the number of pages in a PDF using pypdfium2."""
+    try:
+        pdf = pdfium.PdfDocument(pdf_content)
+        page_count = len(pdf)
+        pdf.close()
+        return page_count
+    except Exception as e:
+        logger.warning(f"Failed to get PDF page count: {e}")
+        return 1  # Assume single page on error
+
+
+# POST /v2/submit_job
+@router.post(
+    "/submit_job",
+    responses={
+        200: {"description": "Jobs were successfully submitted"},
+        500: {"description": "Error encountered while submitting jobs."},
+        503: {"description": "Service unavailable."},
+    },
+    tags=["Ingestion"],
+    summary="submit jobs to the core nv ingestion service for processing with PDF splitting",
+    operation_id="submit_job_v2",
+)
+async def submit_job_v2(request: Request, response: Response, job_spec: MessageWrapper, ingest_service: INGEST_SERVICE_T):
+    with tracer.start_as_current_span("http-submit-job-v2") as span:
```

Comment:
can we turn this tracer logic into a decorator. Hopefully it could be applied as a decorator wherever necessary, given the name is actually the name of the function.